### PR TITLE
♻️ Use denylist

### PR DESCRIPTION
**Pull request by kristoferbaxter**
Move from `blacklist` to `denylist`.

### CODE REVIEW

**Comment by honeybadgerdontcare** [comment]
In validator protoascii we used disallowed, not deny. This change also has internal implications, especially with error messages and other validator engines.

**Comment by kristoferbaxter** [comment]
Going to pull back the changes in protoascii so we can use the standard process to pull changes in.

File: build-system/global-configs/sw-config.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "cache-service-worker-blacklist": [
```
Comment (jridgewell) [inline build-system/global-configs/sw-config.json:2]: I think this file can be deleted entirely. It's an config file for the removed foreign-fetch service worker.

File: extensions/amp-bind/0.1/bind-validator.js [hunk 2]
```diff
@@ -35,7 +35,7 @@ let PropertyRulesDef;
  */
 const GLOBAL_PROPERTY_RULES = {
   'class': {
-    blacklistedValueRegex: '(^|\\W)i-amphtml-',
+    denylistedValueRegex: '(^|\\W)i-amphtml-',
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-validator.js:38]: /cc @choumx I can't tell if this value is actually exposed to publishers. If so, this isn't safe to change.

File: build-system/global-configs/sw-config.json (hunk 1 above)
Comment (kristoferbaxter) [reply build-system/global-configs/sw-config.json:2]: I'll remove it in this PR.

File: extensions/amp-bind/0.1/bind-validator.js (hunk 2 above)
Comment (choumx) [reply extensions/amp-bind/0.1/bind-validator.js:38]: It's not, this was copied from validator code.

File: extensions/amp-form/0.1/form-proxy.js [hunk 3]
```diff
@@ -20,17 +20,17 @@ import {startsWith} from '../../../src/string';
...
-let blacklistedProperties = null;
+let denylistedProperties = null;
 
 /**
  * @param {?Array<string>} properties
  * @visibleForTesting
  */
-export function setBlacklistedPropertiesForTesting(properties) {
-  blacklistedProperties = properties;
+export function setdenylistedPropertiesForTesting(properties) {
```
Comment (choumx) [inline extensions/amp-form/0.1/form-proxy.js:32]: camelCasing

File: third_party/vega/vega.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
...
...=d,t.month=y,t.year=_,t.utcSecond=b,t.utcMinute=x,t.utcHour=w,t.utcDay=k,t.utcSunday=M,t.utcMonday=S,t.utcTuesday=T,t.utcWednesday=E,t.utcThursday=A,t.utcFriday=L,t.utcSaturday=C,t.utcWeek=M,t.utcMonth=D,t.utcYear=P,t.interval=e})},{}],7:[function(t,e,n){var r=t("./util"),i=t("./time"),a=i.utc,o=e.exports;o.$year=r.$func("year",i.year.unit),o.$month=r.$func("month",i.months.unit),o.$date=r.$func("date",i.dates.unit),o.$day=r.$func("day",i.weekdays.unit),o.$hour=r.$func("hour",i.hours.unit),o.$minute=r.$func("minute",i.minutes.unit),o.$second=r.$func("second",i.seconds.unit),o.$utcYear=r.$func("utcYear",a.year.unit),o.$utcMonth=r.$func("utcMonth",a.months.unit),o.$utcDate=r.$func("utcDate",a.dates.unit),o.$utcDay=r.$func("utcDay",a.weekdays.unit),o.$utcHour=r.$func("utcHour",a.hours.unit),o.$utcMinute=r.$func("utcMinute",a.minutes.unit),o.$utcSecond=r.$func("utcSecond",a.seconds.unit)},{"./time":29,"./util":30}],8:[function(t,e,n){function r(){this._cells={},this._aggr=[],this._stream=!1}function i(t){if(a.isArray(t))return t;if(null==t)return[];var e,n,r=[];for(e in t)n=a.array(t[e]),r.push({name:e,ops:n});return r}var a=t("../util"),o=t("./measures"),s=t("./collector"),u=r.Flags={
```
Comment (choumx) [inline third_party/vega/vega.js:1]: We should avoid changing third_party code.

Comment (kristoferbaxter) [reply third_party/vega/vega.js:1]: Yes, this is an unintentional change.

File: extensions/amp-form/0.1/form-proxy.js (hunk 3 above)
Comment (kristoferbaxter) [reply extensions/amp-form/0.1/form-proxy.js]: Good catch! Over eager find/replace looks likely here.

File: extensions/amp-bind/0.1/bind-validator.js (hunk 2 above)
Comment (kristoferbaxter) [reply extensions/amp-bind/0.1/bind-validator.js:38]: It's not safe to change? Or it's not exposed to publishers?

Comment (kristoferbaxter) [reply extensions/amp-bind/0.1/bind-validator.js:38]: Answered offline with @choumx, it's safe to change.

File: test/unit/test-purifier.js [hunk 5]
```diff
@@ -899,9 +899,9 @@ describe('validateAttributeChange', () => {
     expect(vac('p', 'class', 'i-amphtml-illegal')).to.be.false;
     // __amp_source_origin in URLs.
     expect(vac('amp-img', 'src', '?__amp_source_origin=evil')).to.be.false;
-    // BLACKLISTED_TAG_SPECIFIC_ATTRS.
+    // denylistED_TAG_SPECIFIC_ATTRS.
     expect(vac('select', 'form', 'foo')).to.be.false;
-    // BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES.
+    // denylistED_TAG_SPECIFIC_ATTR_VALUES.
```
Comment (choumx) [inline test/unit/test-purifier.js:904]: Nit: casing.

File: src/purifier/sanitation.js [hunk 6]
```diff
@@ -233,10 +233,10 @@ const BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES = Object.freeze(
 );
 
 /**
- * Rules in addition to BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES for AMP4EMAIL.
+ * Rules in addition to denylistED_TAG_SPECIFIC_ATTR_VALUES for AMP4EMAIL.
```
Comment (choumx) [inline src/purifier/sanitation.js:236]: Here too. :)

File: src/purifier/sanitation.js [hunk 7]
```diff
@@ -212,19 +212,19 @@ export const WHITELISTED_ATTRS_BY_TAGS = {
...
-export const WHITELISTED_TARGETS = ['_top', '_blank'];
+export const ALLOWLISTED_TARGETS = ['_top', '_blank'];
 
 // Extended from IS_SCRIPT_OR_DATA in https://github.com/cure53/DOMPurify/blob/master/src/regexp.js.
-const BLACKLISTED_PROTOCOLS = /^(?:\w+script|data|blob):/i;
+const DENYLISTED_PROTOCOLS = /^(?:\w+script|data|blob):/i;
 
-// Same as BLACKLISTED_PROTOCOLS modulo those handled by DOMPurify.
-const EXTENDED_BLACKLISTED_PROTOCOLS = /^(?:blob):/i;
+// Same as denylistED_PROTOCOLS modulo those handled by DOMPurify.
```
Comment (choumx) [inline src/purifier/sanitation.js:220]: ➕

File: test/unit/test-purifier.js (hunk 5 above)
Comment (kristoferbaxter) [reply test/unit/test-purifier.js]: You're good at finding these. Thanks!

**Pull request merged**